public/friendlycode/js/fc/load-project-files.js

```diff
@@ -0,0 +1,215 @@
+define(["jquery"], function($) {
+  "use strict";
+
+  var Path = Bramble.Filer.Path;
+  var FilerBuffer = Bramble.Filer.Buffer;
+
+  // Taken from http://stackoverflow.com/questions/6965107/converting-between-strings-and-arraybuffers
+  function convertToArrayBuffer(string) {
+    var buf = new ArrayBuffer(string.length);
+    var bufView = new Uint8Array(buf);
+    for(var i = 0, strLen = string.length; i < strLen; i++) {
+      bufView[i] = string.charCodeAt(i);
+    }
+    return buf;
+  }
+
+  function load(project, options, callback) {
+    var fs = Bramble.getFileSystem();
+    var shell = new fs.Shell();
+    var projectFilesUrl = window.location.protocol + "//" + window.location.host + "/initializeProject";
+    var request = new XMLHttpRequest();
+    var root = project && project.title;
+    var defaultProject = false;
+    var defaultPath;
+    var files;
+
+    if(!root) {
+      callback(new Error("No project specified"));
+      return;
+    }
+
+    root = Path.join("/", root);
+
+    if(typeof options !== "function") {
+      defaultProject = true;
+      defaultPath = options.isNew ? "/" + project.title : "/New Project";
```

Comment:
How can this be `/New Project`? If `project.title` doesn't exist then it bails on line 29.